pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -132,12 +132,15 @@ func buildContainer(hook *deployapi.ExecNewPodHook, deployment *kapi.Replication
 	// Assigning to a variable since its address is required
 	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds
 
-	podSpec := &kapi.Pod{
+	pod := &kapi.Pod{
 		ObjectMeta: kapi.ObjectMeta{
 			Name: podName,
 			Annotations: map[string]string{
 				deployapi.DeploymentAnnotation: deployment.Name,
 			},
+			Labels: map[string]string{
+				deployapi.DeployerPodForDeploymentLabel: deployment.Name,
```

Comment:
Note to self: this isn't getting applied to the pod for some reason, although it's working for the deployer pod.